compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4182,6 +4182,8 @@ public void visit(BLangTypeInit cIExpr, AnalyzerData data) {
         if (actualType == symTable.semanticError) {
             //TODO dlog error?
             data.resultType = symTable.semanticError;
+            // Following line checks the object compatibility if there is a semantic error.
+            checkObjectCompatibility(actualType, cIExpr, data);
```

Comment:
As we discussed offline, we need to add more test cases to cover scenarios enabled by this fix. As this can lead to Bad Sad errors.  
Solution for failing test cases -> Check SemanticError type and gracefully recover from that error.